src/C/common.jl

```diff
@@ -55,35 +48,33 @@ const CXT_Comment = (UInt32)(3)
 const CXT_Literal = (UInt32)(4)
 # end enum ANONYMOUS_2
 
-# begin enum CPLXMLNodeType
-const CPLXMLNodeType = UInt32
-const CXT_Element = (UInt32)(0)
-const CXT_Text = (UInt32)(1)
-const CXT_Attribute = (UInt32)(2)
-const CXT_Comment = (UInt32)(3)
-const CXT_Literal = (UInt32)(4)
-# end enum CPLXMLNodeType
+const CPLXMLNodeType = Void
 
-type CPLXMLNode
+mutable struct CPLXMLNode
     eType::CPLXMLNodeType
     pszValue::Cstring
-    psNext::Ptr{CPLXMLNode}
-    psChild::Ptr{CPLXMLNode}
+    psNext::Ptr{Void}
+    psChild::Ptr{Void}
 end
```

Comment:
I'm not sure about recursive types like these, since I haven't yet used and/or tested them

Reply:
Yeah me neither, I'll try them out / add a test before merging.

Reply:
In the latest wrapping it was no longer self referential, but I added a test in 57dac89c11fc16ccb2bab7d3aa9ddcf307a97a92